gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -182,6 +183,19 @@ private static void updateBasedOnReferenceToReferenceDuplicatedPlan(List<Referen
         return duplicatedMedicationStatement;
     }
 
+    private static void updateDuplicatedMedicationStatementEffectivePeriod(
+        MedicationStatement duplicatedMedicationStatement,
+        MedicationRequest order
+
+    ) {
+        duplicatedMedicationStatement.setEffective(order.getDispenseRequest().getValidityPeriod().copy());
+
+        var effectivePeriod = duplicatedMedicationStatement.getEffectivePeriod();
+        if (!ACTIVE.equals(duplicatedMedicationStatement.getStatus()) && !effectivePeriod.hasEnd()) {
+            effectivePeriod.setEndElement(effectivePeriod.getStartElement());
+        }
```

Comment:
I'm getting a strong stench that the software design is a bit off now that we have behaviour duplicated between here and `MedicationStatementMapper::mapToMedicationStatement`.

Thoughts?

Reply:
Yup I feel this also, it seems like maybe this whole functionality should sit in the MedicationStatementMapper, although it's not really mapping as such.  It could be moved there, but it would be a change in scope as the mappers generally take in the ehr components.. hmmm...

Reply:
This, and the medication request plan will be moved into the associated mappers